src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.CodeFixes/ApiControllerCodeFixer.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.CodeFixes
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = "UA0013 CodeFix Provider")]
+    public class ApiControllerCodeFixer : CodeFixProvider
+    {
+        public const string GoodNamespace = "Microsoft.AspNetCore.Mvc";
+        public const string GoodClassName = "Controller";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(ApiControllerAnalyzer.DiagnosticId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root is null)
+            {
+                return;
+            }
+
+            var node = root.FindNode(context.Span);
+
+            if (node.IsKind(CodeAnalysis.CSharp.SyntaxKind.SimpleBaseType))
+            {
+                // When the analyzer runs it selects a portion of the code to be fixed by this class.
+                // In CSharp syntax, the SimpleBaseTypeSyntax is at the same location as the QualifiedNameSyntax or IdentifierNameSyntax
+                // since we just found this node by text location, we need to adjust what we selected to match our assumption for CSharp
+                node = node.DescendantNodes().First();
```

Comment:
was looking for, but did not find, a way to pass a reference to the SyntaxNode from the Analyzer to the Code Fixer. the only overloads I saw were location based details.